x/leverage/keeper/keeper_test.go

```diff
@@ -128,6 +147,314 @@ func (suite *IntegrationTestSuite) TestWithdrawAsset_Valid() {
+	err = app.LeverageKeeper.RepayAsset(ctx,
+		lenderAddr,
+		tCoin("umee", 300),
+	)
+	suite.Require().NoError(err)
+
+	// verify lender's new loan amount is zero
+	loanBalance := app.LeverageKeeper.GetLoan(ctx, lenderAddr, umeeapp.BondDenom)
+	suite.Require().Equal(loanBalance, tCoin("umee", 0))
+
+	// verify lender's new umee balance (10 - 1k from initial + 200 from loan - 200 repaid = 9000)
+	tokenBalance := app.BankKeeper.GetBalance(ctx, lenderAddr, umeeapp.BondDenom)
+	suite.Require().Equal(tokenBalance, tCoin("umee", 9000))
+
+	// verify lender's uToken balance remains at 1k from initial conditions
+	uTokenBalance := app.BankKeeper.GetBalance(ctx, lenderAddr, "u/"+umeeapp.BondDenom)
+	suite.Require().Equal(uTokenBalance, tCoin("u/umee", 1000))
+
+	// lender repays 50 umee - this time it fails because the loan no longer exists
+	err = app.LeverageKeeper.RepayAsset(ctx,
+		lenderAddr,
+		tCoin("umee", 50),
+	)
+	suite.Require().Error(err)
+}
+
+func (suite *IntegrationTestSuite) TestSetCollateral_Valid() {
+	lenderAddr, _ := suite.initBorrowScenario() // create initial conditions
+	app, ctx := suite.app, suite.ctx            // get ctx after init
+
+	// Any user from the starting scenario can be used, since they are only toggling
+	// collateral settings.
+
+	// lender disables u/umee as collateral
+	err := app.LeverageKeeper.SetCollateralSetting(ctx,
+		lenderAddr,
+		"u/uumee",
+		false,
+	)
+	suite.Require().NoError(err)
+	enabled := app.LeverageKeeper.GetCollateralSetting(ctx,
+		lenderAddr,
+		"u/uumee",
+	)
+	suite.Require().Equal(enabled, false)
+
+	// lender enables u/umee as collateral
+	err = app.LeverageKeeper.SetCollateralSetting(ctx,
+		lenderAddr,
+		"u/uumee",
+		true,
+	)
+	suite.Require().NoError(err)
+	enabled = app.LeverageKeeper.GetCollateralSetting(ctx,
+		lenderAddr,
+		"u/uumee",
+	)
+	suite.Require().Equal(enabled, true)
+}
+
+func (suite *IntegrationTestSuite) TestSetCollateral_Invalid() {
+	lenderAddr, _ := suite.initBorrowScenario() // create initial conditions
+	app, ctx := suite.app, suite.ctx            // get ctx after init
+
+	// Any user from the starting scenario can be used, since they are only toggling
+	// collateral settings.
+
+	// lender disables u/abcd as collateral - fails because "u/abcd" is not a recognized uToken
+	err := app.LeverageKeeper.SetCollateralSetting(ctx,
+		lenderAddr,
+		"u/abcd",
+		false,
+	)
+	suite.Require().Error(err)
+
+	// lender disables uumee as collateral - fails because "uumee" is an asset, not a uToken
+	err = app.LeverageKeeper.SetCollateralSetting(ctx,
+		lenderAddr,
+		"uumee",
+		false,
+	)
+	suite.Require().Error(err)
+}
+
+func (suite *IntegrationTestSuite) TestGetCollateral_Invalid() {
+	lenderAddr, _ := suite.initBorrowScenario() // create initial conditions
+	app, ctx := suite.app, suite.ctx            // get ctx after init
+
+	// Any user from the starting scenario can be used, since we are only viewing
+	// collateral settings.
```

Comment:
nit: this applies to all the tests. Try to keep things on one line if they can fit under 120 chars. Also, avoid redundant comments.